### PR TITLE
Modular hawkBit

### PR DESCRIPTION
**Pull request by kaizimmerm**
I created spring boot starters for hawkBit in general and separate ones for the four interfaces. That allows application builders to split hawkBit functionality into separate services.

While doing that I fixed a lot of bugs around component scan. It was effectively not use single components of hawkBit and still leverage the auto configuration package. I think I found all issues and at least the split that is proposed with the handy starters work.

I removed the Enable* annotations as they never worked anyway.

Reviewers: @SirWayne @Jonkno

### CODE REVIEW

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/UploadArtifactView.java [hunk 1]
```diff
@@ -92,7 +92,7 @@
             final UINotification uiNotification, final ArtifactUploadState artifactUploadState,
             final TagManagement tagManagement, final EntityFactory entityFactory,
             final SoftwareManagement softwareManagement, final UploadViewClientCriterion uploadViewClientCriterion,
-            final SPInfo spInfo, final ArtifactManagement artifactManagement) {
+            final MultipartConfigElement spInfo, final ArtifactManagement artifactManagement) {
```
Comment (SirWayne) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/UploadArtifactView.java]: I would rename ```spInfo```

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/upload/UploadLayout.java [hunk 2]
```diff
@@ -89,7 +90,7 @@
 
     private final ArtifactUploadState artifactUploadState;
 
-    private final transient SPInfo spInfo;
+    private final transient MultipartConfigElement spInfo;
```
Comment (SirWayne) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/upload/UploadLayout.java]: I would rename ```spInfo```

File: hawkbit-autoconfigure/src/main/java/org/eclipse/hawkbit/autoconfigure/security/SecurityManagedConfiguration.java [hunk 3]
```diff
@@ -151,6 +156,26 @@ private UserAuthenticationFilterBasicAuth(final AuthenticationManager authentica
...
         private SystemSecurityContext systemSecurityContext;
 
+        /**
+         * Filter to protect the hawkBit server DDI interface against to many
+         * requests.
+         * 
+         * @param securityProperties
+         *            for filter configuration
+         *
+         * @return he spring filter registration bean for registering an denial
```
Comment (Jonkno) [inline hawkbit-autoconfigure/src/main/java/org/eclipse/hawkbit/autoconfigure/security/SecurityManagedConfiguration.java]: Typos: @return the ... a detail

**Comment by Jonkno** [comment]
I would prefer to also have README.md files for the Spring boot starters

File: hawkbit-autoconfigure/src/main/resources/META-INF/spring.factories [hunk 4]
```diff
@@ -5,11 +5,14 @@ org.eclipse.hawkbit.autoconfigure.ui.UIAutoConfiguration,\
...
 org.eclipse.hawkbit.autoconfigure.cache.CacheAutoConfiguration,\
 org.eclipse.hawkbit.autoconfigure.cache.DownloadIdCacheAutoConfiguration,\
 org.eclipse.hawkbit.autoconfigure.event.EventPublisherAutoConfiguration,\
 org.eclipse.hawkbit.autoconfigure.scheduling.AsyncConfigurerAutoConfiguration,\
 org.eclipse.hawkbit.autoconfigure.scheduling.ExecutorAutoConfiguration,\
-org.eclipse.hawkbit.autoconfigure.amqp.AmqpAutoConfiguration,\
+org.eclipse.hawkbit.autoconfigure.amqp.DmfApiAutoConfiguration,\
 org.eclipse.hawkbit.autoconfigure.security.InMemoryUserManagementConfiguration,\
-org.eclipse.hawkbit.autoconfigure.repository.ArtifactStoreAutoConfiguration
+org.eclipse.hawkbit.autoconfigure.repository.ArtifactStoreAutoConfiguration,\
```
Comment (SirWayne) [inline hawkbit-autoconfigure/src/main/resources/META-INF/spring.factories]: Change to ```ArtifactFilesystemAutoConfiguration ```

File: hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/exception/MethodNotSupported.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+ *
+ */
+public final class MethodNotSupported extends AbstractServerRtException {
+    /**
+    *
+    */
+    private static final long serialVersionUID = 1L;
+
+    /**
+     * Creates a new FileUploadFailedException with
```
Comment (SirWayne) [inline hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/exception/MethodNotSupported.java]: The javadoc is wrong --> it's not a ```FileUploadFailedException```

**Pull request merged**